### ISSUE TITLE
Too many warnings for unused definition

### ISSUE THREAD

**Issue by mohsen1** [apigee-127/swagger-tools#201]
With this Swagger I get 3 warnings for unused schema. It should be one

```yaml
---
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
    get:
      responses:
        200:
          description: OK
definitions:
  User:
    allOf:
      - 
        type: string
    
      - 
        enum: ['One']
      
```

**Comment by whitlockjc** [comment]
Which version?  I just fixed this...or so I thought.

**Comment by mohsen1** [comment]
The editor is using the latest.

**Comment by whitlockjc** [comment]
I'm not seeing this.  If I run the `swagger-tools validate swagger.yaml`, I get this:

```
swagger-tools validate swagger.yaml
API Warnings:

  #/definitions/User: Definition is defined but is not used: #/definitions/User

Validation succeeded but with 1 warning
```

Is there any chance you've got a cache in place that might be creating this?  I know with v0.8.5 this would likely happen since inline schemas in `allOf` were marked as unused but that was fixed with v0.8.6.

**Comment by mohsen1** [comment]
Just tried again in the latest master of Swagger Editor which is using `.8.6` from Bower and got the 3 errors with that spec. Maybe it's a browser/node inconsistency?

**Comment by whitlockjc** [comment]
We have tests that run in browser and none of them report failures more than once, or all the browser tests would fail.  All tests in `test/1.2/test-specs.js` and `test/2.0/test-specs.js` run in the browser and all tests in there are very explicit about what is expected, none of which is three versions of the same error.

I'm not disagreeing that it's happening so I'll look further I'm just not seeing it just yet outside of swagger-editor.  I'll do some manual testing and see what I find.  When I have something, I'll report here.

**Comment by mohsen1** [comment]
This snippet inside Swagger Editor console is logging 3 errors. You can try it in your manual testing:


```js
SwaggerTools.specs.v2.validate ({
    "swagger": "2.0",
    "info": {
        "version": "0.0.0",
        "title": "Simple API"
    },
    "paths": {
        "/": {
            "get": {
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    },
    "definitions": {
        "User": {
            "allOf": [
                {
                    "type": "string"
                },
                {
                    "enum": [
                        "One"
                    ]
                }
            ]
        }
    }
}, function(){console.log(arguments)})
```

**Comment by whitlockjc** [comment]
Thanks.  I'll do some playing around.

**Comment by whitlockjc** [comment]
I wonder if the browser build was not added to the release...I'll check.

**Comment by whitlockjc** [comment]
hahahahaha...v0.8.6 does not have this fix in it: https://github.com/apigee-127/swagger-tools/commits/master

It seems that I fixed this _after_ v0.8.6 was released.  That would do it.  I'm marking this as a duplicate of #187.
